### PR TITLE
Cleanup of Catch[CN]-related restart utilities in ./GEOSsurface_GridComp/Utils/mk_restarts

### PR DESCRIPTION
**Pull request by gmao-rreichle**
This pull request addresses a major overhaul of the mk restarts utilities, with a focus on Catch[CN] restarts.  

I started a branch specifically for this PR that already includes a few minor corrections and in-line comments.  Feel free to commit to this branch or make sure to stay in sync with it. 

Here's a first draft of the task list (not necessarily in the order in which tasks are best addressed):

1. Clean up duplication of code across "mk Catch[CN] restarts" utilities:
    - mk_GEOSldasRestarts.F90 and mk_Catch[CN]Restarts.F90 -- Merge into one program? (Probably need model-specific modules.)
    - mk_catch_restart, mk_catch_restart.F90 -- Are these obsolete?  (A quick check suggests this is not used in GEOSgcm_GridComp, GMAO_Shared, GEOSgcm[_App], or GEOSldas, but need to verify.)
    - Scale_Catch.F90 and Scale_CatchCN.F90 -- Merge into one program?  (Probably need model-specific modules.)
    - To simplify things, it's probably best to **drop support for LDASsa restarts.**

2. Interface new "mk_CatchX_Restarts" utility with: 
    - GMAO_Shared/GEOS_Util/post/regrid.p[l,y] (perl and/or python version, depending on status of GMAO_Shared)
    - GEOSldas setup
    - GCM setup (?) - may be addressed through "regrid.p[l,y]"

3. Remove obsolete code (need to verify that code is indeed obsolete):
    - replace_params.F90 
    - strip_vedyn.F90
    - newvegdyn.F90
    - newcatch.F90 (??)
    - old_catch.ctl -- "Old" here likely refers to a very old format and tile space.  Note hardcoded number of catchments.
    - new_catch.ctl -- Note hardcoded number of catchments. Hardcoded list of variables is probably outdated or too inflexible (e.g., TSURF was removed from GEOSldas restarts).  
    
4. Move potentially still useful plotting scripts into new sub-directory:
    - check_land_restarts.pro -- Would this still work? If not, delete.
    - catchplt -- Obsolete? If yes, delete.

5. Address duplication of subroutines across ./Utils/Raster/ and ./Utils/mk_restarts/ (and possibly elsewhere)
    - haversine()
    - to_radians()
    - ReadCNTilFile() [comp_CATCHCN_AlbScale_parameters.F90] and ReadTileFile_RealLatLon()  [getids.F90]

6. Update README file.

7. Clean up `.gitignore` ?

For reference, here's a list of related pull requests and issues:

Open:

- https://github.com/GEOS-ESM/GEOSldas/issues/360
- #344 (may have been addressed)
- #116 (presumably addressed fully for Catch, not sure about CatchCN)
- https://github.com/GEOS-ESM/GMAO_Shared/pull/238
- https://github.com/GEOS-ESM/GMAO_Shared/pull/257
 

Merged:

- #402
- #368 
- #224 

cc: @gmao-jkolassa @rdkoster @biljanaorescanin @weiyuan-jiang

### CODE REVIEW

**Comment by gmao-rreichle** [comment]
In line 281 of mk_Restarts, the CatchCN restart file name is referenced as "catchcn_internal_rst", but didn't we make the CatchCN restart file names specific to the CatchCN version (ie, CatchCNCLM40 and CatchCNCLM45)?
https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/d59a8154dd5a772aaaa8997ef9fbe1a7230ad88b/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/mk_Restarts#L281
Presumably, the mk_Restarts file would fail if we try to regrid (re-tile) a newer CatchCN restart file.

**Comment by gmao-rreichle** [comment]
In line 60 of mk_CatchRestarts.F90, getarg() is used to read command line argument number 6, but earlier in the code the number of command arguments was constrained to 4 or 5. 
https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/d59a8154dd5a772aaaa8997ef9fbe1a7230ad88b/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/mk_CatchRestarts.F90#L60
I'm probably misunderstanding something here but thought I should flag this, just in case.

**Comment by weiyuan-jiang** [comment]
> In line 281 of mk_Restarts, the CatchCN restart file name is referenced as "catchcn_internal_rst", but didn't we make the CatchCN restart file names specific to the CatchCN version (ie, CatchCNCLM40 and CatchCNCLM45)?
> 
> https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/d59a8154dd5a772aaaa8997ef9fbe1a7230ad88b/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/mk_Restarts#L281
> 
> Presumably, the mk_Restarts file would fail if we try to regrid (re-tile) a newer CatchCN restart file.

That line was used for very old cn. Now cnlen is always 4 in the regrid.pl

**Comment by weiyuan-jiang** [comment]
> In line 60 of mk_CatchRestarts.F90, getarg() is used to read command line argument number 6, but earlier in the code the number of command arguments was constrained to 4 or 5.
> 
> https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/d59a8154dd5a772aaaa8997ef9fbe1a7230ad88b/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/mk_CatchRestarts.F90#L60
> 
> I'm probably misunderstanding something here but thought I should flag this, just in case.

I think it is just for old version of the output ( read or write)

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/CatchmentCNRst.F90 [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+       catch_calc_tp,                   &
+       catch_calc_ght
+  USE CATCH_CONSTANTS,   ONLY:          &
+       N_GT              => CATCH_N_GT, &
+       DZGT              => CATCH_DZGT, &
+       PEATCLSM_POROS_THRESHOLD
+  use CatchmentRstMod, only : CatchmentRst
+  implicit none
+
+  integer, parameter :: nveg  =  integer, parameter :: nveg    = 4
```
Comment (gmao-rreichle) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/CatchmentCNRst.F90:16]: Looks like a cut-and-paste error here. 
(I just stumbled across this, not doing a thorough code review.)

**Comment by weiyuan-jiang** [comment]
I got zero-diff from M36 to M09 for catchment. That is a good start. Will try cnclm45 @gmao-rreichle

**Review by sdrabenh — APPROVED** [review]

**Review by biljanaorescanin — APPROVED** [review]

**Review by bena-nasa — APPROVED** [review]

**Pull request merged**